### PR TITLE
GPII-3810: Onboard Windows 10 "Mirror Screen"

### PR DESCRIPTION
**Pull request by JavierJF**

### CODE REVIEW

File: gpii/node_modules/nativeSettingsHandler/src/nativeSettingsHandler.js [hunk 1]
```diff
@@ -39,6 +39,79 @@ windows.nativeSettingsHandler.GetDoubleClickTime = function () {
...
+ *
+ * Note:
+ * The native function 'SetDisplayConfig' returns the error 'ERROR_GEN_FAILURE' which stands to code
+ * 31 when a second display isn't present, this error may be misleading because no system component
+ * is malfunctioning.
+ *
+ * @param {Boolean} bool A boolean representing the desired state for the 'screen mirror' feature.
+ * @return {Promise} A promise that resolves on success or holds a error code and message on failure.
+ */
+windows.nativeSettingsHandler.SetScreenMirrored = function (bool) {
```
Comment (amb26) [inline gpii/node_modules/nativeSettingsHandler/src/nativeSettingsHandler.js]: Please give semantic name to "bool"

File: gpii/node_modules/nativeSettingsHandler/src/nativeSettingsHandler.js [hunk 2]
```diff
@@ -39,6 +39,79 @@ windows.nativeSettingsHandler.GetDoubleClickTime = function () {
...
+    var pRes = fluid.promise();
+    var err = 0;
+
+    if (bool) {
+        var cloneFlags = windows.API_constants.SDC_TOPOLOGY_CLONE | windows.API_constants.SDC_APPLY;
```
Comment (amb26) [inline gpii/node_modules/nativeSettingsHandler/src/nativeSettingsHandler.js]: Please fold these near-duplicates into two lines with ternaries

File: gpii/node_modules/nativeSettingsHandler/src/nativeSettingsHandler.js (hunk 1 above)
Comment (JavierJF) [reply gpii/node_modules/nativeSettingsHandler/src/nativeSettingsHandler.js]: Solved!

File: gpii/node_modules/nativeSettingsHandler/src/nativeSettingsHandler.js (hunk 2 above)
Comment (JavierJF) [reply gpii/node_modules/nativeSettingsHandler/src/nativeSettingsHandler.js]: Done!

**Comment by JavierJF** [comment]
Looks like fail was related to not being able to close a process:

image

Nothing that has to do with pull request contents. Testing again.

**Comment by JavierJF** [comment]
Ok to test

**Comment by stegru** [comment]
Increasing the timeout here may help: https://github.com/GPII/windows/blob/master/gpii/node_modules/processHandling/test/testProcessHandling.js#L186-L186

1 second is probably not enough to: start a .net application, create a window, read the output of it, then send and process a window message.

(it seems like I missed a `0`)

**Comment by JavierJF** [comment]
Last error looks like an out of sync problem with universal, due to both builds being triggered at almost the same time:

image

Retrying again.

**Comment by JavierJF** [comment]
Ok to test

**Comment by amb26** [comment]
Hi there @JavierJF - I've tried this against the latest universal release which incorporates all the related solutions registry pulls 
https://github.com/GPII/universal/pull/624
https://github.com/GPII/universal/pull/758
https://github.com/GPII/universal/pull/760

and got the following mangled exception report on the acceptance tests:
````
06:00:37.906:  ASSERTION FAILED: Error when operating settings handler: {
    "0": "C",
    "1": "a",
    "2": "n",
    "3": "n",
    "4": "o",
    "5": "t",
    "6": " ",
    "7": "c",
    "8": "r",
    "9": "e",
    "10": "a",
    "11": "t",
    "12": "e",
    "13": " ",
    "14": "s",
    "15": "t",
    "16": "r",
    "17": "u",
    "18": "c",
    "19": "t",
    "20": "u",
    "21": "r",
    "22": "e",
    "23": " ",
    "24": "T",
    "25": "O",
    "26": "G",
    "27": "G",
    "28": "L",
    "29": "E",
    "30": "K",
    "31": "E",
    "32": "Y",
    "33": "S",
    "34": " ",
    "35": "-",
    "36": " ",
    "37": "p",
    "38": "a",
    "39": "y",
    "40": "l",
    "41": "o",
    "42": "a",
    "43": "d",
    "44": ".",
    "45": "o",
    "46": "p",
    "47": "t",
    "48": "i",
    "49": "o",
    "50": "n",
    "51": "s",
    "52": ".",
    "53": "p",
    "54": "v",
    "55": "P",
    "56": "a",
    "57": "r",
    "58": "a",
    "59": "m",
    "60": ".",
    "61": "t",
    "62": "e",
    "63": "m",
    "64": "p",
    "65": "l",
    "66": "a",
    "67": "t",
    "68": "e .... [output suppressed at 1024 chars - for more output, increase fluid.logObjectRenderChars]
06:00:37.906:  jq: FAIL: Module "gpii.tests.acceptance.windows.builtIn.config tests" Test name "Testing os_win using default matchmaker" - Message: Assertion failure (see console.log for expanded message): Error when operating settings handler: ,[object Object]
06:00:37.906:  jq: Source:     at jqUnit.nodeFailureHandler (V:\node_modules\node-jqunit\lib\jqUnit-node.js:34:15)
    at Object.fire (V:\node_modules\infusion\src\framework\core\js\Fluid.js:1612:40)
    at Object.fluid.fail (V:\node_modules\infusion\src\framework\core\js\Fluid.js:222:32)
    at V:\node_modules\gpii-universal\gpii\node_modules\testing\src\TestUtils.js:143:15
    at Object.that.then (V:\node_modules\infusion\src\framework\core\js\FluidPromises.js:40:21)
    at Object.gpii.test.operateSettings (V:\node_modules\gpii-universal\gpii\node_modules\testing\src\TestUtils.js:142:10)
    at Object.gpii.test.getSettings (V:\node_modules\gpii-universal\gpii\node_modules\testing\src\TestUtils.js:114:22)
    at gpii.test.snapshotSettings (V:\node_modules\gpii-universal\gpii\node_modules\testing\src\TestUtils.js:166:33)
    at Object.expanded.apply (V:\node_modules\infusion\tests\test-core\utils\js\IoCTestUtils.js:171:38)
    at Object.execute (V:\node_modules\infusion\tests\test-core\utils\js\IoCTestUtils.js:228:24)
    at Ob .... [output suppressed at 1024 chars - for more output, increase fluid.logObjectRenderChars]
06:00:37.923:  FATAL ERROR: Uncaught exception: Assertion failure - check console for more details: Error when operating settings handler: {
    "0": "C",
    "1": "a",
    "2": "n",
    "3": "n",
    "4": "o",
    "5": "t",
    "6": " ",
    "7": "c",
    "8": "r",
    "9": "e",
    "10": "a",
    "11": "t",
    "12": "e",
    "13": " ",
    "14": "s",
    "15": "t",
    "16": "r",
    "17": "u",
    "18": "c",
    "19": "t",
    "20": "u",
    "21": "r",
    "22": "e",
    "23": " ",
    "24": "T",
    "25": "O",
    "26": "G",
    "27": "G",
    "28": "L",
    "29": "E",
    "30": "K",
    "31": "E",
    "32": "Y",
    "33": "S",
    "34": " ",
    "35": "-",
    "36": " ",
    "37": "p",
    "38": "a",
    "39": "y",
    "40": "l",
    "41": "o",
    "42": "a",
    "43": "d",
    "44": ".",
    "45": "o",
    "46": "p",
    "47": "t",
    "48": "i",
    "49": "o",
    "50": "n",
    "51": "s",
    "52": ".",
    "53": "p",
    "54": "v",
    "55": "P",
    "56": "a",
    "57": "r",
    "58": "a",
    "59": "m",
    "60" .... [output suppressed at 1024 chars - for more output, increase fluid.logObjectRenderChars]
Error: Assertion failure - check console for more details: Error when operating settings handler: {
    "0": "C",
    "1": "a",
    "2": "n",
    "3": "n",
    "4": "o",
    "5": "t",
    "6": " ",
    "7": "c",
    "8": "r",
    "9": "e",
    "10": "a",
    "11": "t",
    "12": "e",
    "13": " ",
    "14": "s",
    "15": "t",
    "16": "r",
    "17": "u",
    "18": "c",
    "19": "t",
    "20": "u",
    "21": "r",
    "22": "e",
    "23": " ",
    "24": "T",
    "25": "O",
    "26": "G",
    "27": "G",
    "28": "L",
    "29": "E",
    "30": "K",
    "31": "E",
    "32": "Y",
    "33": "S",
    "34": " ",
    "35": "-",
    "36": " ",
    "37": "p",
    "38": "a",
    "39": "y",
    "40": "l",
    "41": "o",
    "42": "a",
    "43": "d",
    "44": ".",
    "45": "o",
    "46": "p",
    "47": "t",
    "48": "i",
    "49": "o",
    "50": "n",
    "51": "s",
    "52": ".",
    "53": "p",
    "54": "v",
    "55": "P",
    "56": "a",
    "57": "r",
    "58": "a",
    "59": "m",
    "60": ".",
    "61": "t",
    "62": "e",
    "63": "m",
    "64": "p",
    "65": "l",
    "66": "a",
    "67": "t",
    "68": "e .... [output suppressed at 1024 chars - for more output, increase fluid.logObjectRenderChars]
    at new fluid.FluidError (V:\node_modules\infusion\src\framework\core\js\Fluid.js:181:26)
    at Object.fluid.builtinFail (V:\node_modules\infusion\src\framework\core\js\Fluid.js:205:15)
    at jqUnit.nodeFailureHandler (V:\node_modules\node-jqunit\lib\jqUnit-node.js:38:11)
    at Object.fire (V:\node_modules\infusion\src\framework\core\js\Fluid.js:1612:40)
    at Object.fluid.fail (V:\node_modules\infusion\src\framework\core\js\Fluid.js:222:32)
    at V:\node_modules\gpii-universal\gpii\node_modules\testing\src\TestUtils.js:143:15
    at Object.that.then (V:\node_modules\infusion\src\framework\core\js\FluidPromises.js:40:21)
    at Object.gpii.test.operateSettings (V:\node_modules\gpii-universal\gpii\node_modules\testing\src\TestUtils.js:142:10)
    at Object.gpii.test.getSettings (V:\node_modules\gpii-universal\gpii\node_modules\testing\src\TestUtils.js:114:22)
    at gpii.test.snapshotSettings (V:\node_modules\gpii-universal\gpii\node_modules\testing\src\TestUtils.js:166:33)
    at Object.expanded.apply (V:\node_modules\infusion\tests\test-core\utils\js\IoCTestUtils.js:171:38)
    at Object.execute (V:\node_modules\infusion\tests\test-core\utils\js\IoCTestUtils.js:228:24)
    at Object.fluid.test.execExecutor (V:\node_modules\infusion\tests\test-core\utils\js\IoCTestUtils.js:506:18)
    at that.execute (V:\node_modules\infusion\tests\test-core\utils\js\IoCTestUtils.js:644:32)
    at Object.execute (V:\node_modules\infusion\tests\test-core\utils\js\IoCTestUtils.js:229:17)
    at Object.fluid.test.execExecutor (V:\node_modules\infusion\tests\test-core\utils\js\IoCTestUtils.js:506:18)
    at that.execute (V:\node_modules\infusion\tests\test-core\utils\js\IoCTestUtils.js:647:32)
    at Object.execute (V:\node_modules\infusion\tests\test-core\utils\js\IoCTestUtils.js:229:17)
    at Object.fluid.test.execExecutor (V:\node_modules\infusion\tests\test-core\utils\js\IoCTestUtils.js:506:18)
    at that.execute (V:\node_modules\infusion\tests\test-core\utils\js\IoCTestUtils.js:647:32)
    at V:\node_modules\infusion\tests\test-core\utils\js\IoCTestUtils.js:310:13
    at wrapped (V:\node_modules\infusion\tests\test-core\utils\js\IoCTestUtils.js:321:17)
    at fire (V:\node_modules\infusion\src\framework\core\js\Fluid.js:1612:40)
    at V:\node_modules\gpii-universal\gpii\node_modules\testing\src\TestUtils.js:188:9
    at Object.that.then (V:\node_modules\infusion\src\framework\core\js\FluidPromises.js:33:21)
    at gpii.test.checkConfiguration (V:\node_modules\gpii-universal\gpii\node_modules\testing\src\TestUtils.js:185:19)
    at Object.expanded.apply (V:\node_modules\infusion\tests\test-core\utils\js\IoCTestUtils.js:171:38)
    at Object.execute (V:\node_modules\infusion\tests\test-core\utils\js\IoCTestUtils.js:228:24)
    at Object.fluid.test.execExecutor (V:\node_modules\infusion\tests\test-core\utils\js\IoCTestUtils.js:506:18)
    at that.execute (V:\node_modules\infusion\tests\test-core\utils\js\IoCTestUtils.js:644:32)
    at V:\node_modules\infusion\tests\test-core\utils\js\IoCTestUtils.js:310:13
    at wrapped (V:\node_modules\infusion\tests\test-core\utils\js\IoCTestUtils.js:321:17)
    at Array.fire (V:\node_modules\infusion\src\framework\core\js\Fluid.js:1612:40)
    at Object.that.complete (V:\node_modules\infusion\src\framework\core\js\FluidPromises.js:70:25)
    at Object.that.resolve (V:\node_modules\infusion\src\framework\core\js\FluidPromises.js:52:22)
    at Object.resolve (V:\node_modules\gpii-universal\gpii\node_modules\testing\src\TestUtils.js:43:28)
    at Array.<anonymous> (V:\node_modules\gpii-universal\gpii\node_modules\testing\src\TestUtils.js:78:23)
    at Object.that.complete (V:\node_modules\infusion\src\framework\core\js\FluidPromises.js:70:25)
    at Object.that.resolve (V:\node_modules\infusion\src\framework\core\js\FluidPromises.js:52:22)
    at Array.<anonymous> (V:\gpii\node_modules\registeredAT\registeredAT.js:60:187)
    at Object.that.complete (V:\node_modules\infusion\src\framework\core\js\FluidPromises.js:70:25)
    at Object.that.resolve (V:\node_modules\infusion\src\framework\core\js\FluidPromises.js:52:22)
    at Function.fluid.promise.resumeSequence (V:\node_modules\infusion\src\framework\core\js\FluidPromises.js:163:26)
    at Function.fluid.promise.progressSequence (V:\node_modules\infusion\src\framework\core\js\FluidPromises.js:149:23)
    at Array.<anonymous> (V:\node_modules\infusion\src\framework\core\js\FluidPromises.js:169:35)
    at Object.that.complete (V:\node_modules\infusion\src\framework\core\js\FluidPromises.js:70:25)
    at Object.that.resolve (V:\node_modules\infusion\src\framework\core\js\FluidPromises.js:52:22)
    at V:\gpii\node_modules\registeredAT\registeredAT.js:58:445
    at ChildProcess.exithandler (child_process.js:280:7)
    at ChildProcess.emit (events.js:182:13)
    at maybeClose (internal/child_process.js:962:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:251:5)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! gpii-windows@0.3.0 test:acceptance: `set GPII_TEST_COUCH_USE_EXTERNAL=TRUE && nyc node.exe ./tests/AcceptanceTests.js builtIn`
npm ERR! Exit status 1
````
It looks like it has attempted to convert the string "Cannot create structure TOGGLEKEYS" directly into a structure. Could you please fix the exception handler which converts this messages as well as the underlying issue - thanks!

**Pull request merged**